src/services/user.service.js

```diff
@@ -4,6 +4,38 @@ const { userRepo, tokenRepo } = require('../models/repo')
 const db = require('../models')
 const { generateAuthTokens, verifyRefreshToken } = require('./token.service')
 const { tokenTypes } = require('../config/tokens.config')
+const { hashPassword } = require('../utils')
+
+/**
+ * Change user password
+ * @param {Object} userObject
+ * @param {id} userObject.userId
+ * @param {string} userObject.currentPassword
+ * @param {string} userObject.newPassword
+ * @throws {NotFoundError} if user not exist
+ * @throws {AuthFailureError} if password not match
+ * @throws {BadRequestError} if update password failed
+ */
+const changePassword = async ({ userId, currentPassword, newPassword }) => {
+    const user = await userRepo.getUserById(userId)
+    if (!user) {
+        throw new NotFoundError('Not found user')
+    }
+
+    const isMatchPassword = await bcrypt.compare(currentPassword, user.password)
+    if (!isMatchPassword) {
+        throw new AuthFailureError('Incorrect current password')
+    }
+    if (newPassword === currentPassword) {
+        throw new BadRequestError('The new password cannot be the same as the old one')
```

Comment:
"New Password cannot be same as your current password. Please choose a different password."

Reply:
done